### PR TITLE
[API stabilization] Rework JSON DSL API

### PR DESCRIPTION
**Pull request by qwwdfsad**
* Make it aligned with Kotlin standard library where receiver of buildList/buildMap is MutableList/MutableMap
    * Make it aligned with our own new naming scheme where all JSON subtypes are mentioned explicitly

### CODE REVIEW

File: runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt [hunk 1]
```diff
@@ -147,99 +117,143 @@ public class JsonObjectBuilder internal constructor() {
...
-    private val content: MutableList<JsonElement> = mutableListOf()
+public fun JsonObjectBuilder.putJsonObject(key: String, builderAction: JsonObjectBuilder.() -> Unit): JsonElement? =
+    put(key, buildJsonObject(builderAction))
 
+/**
+ * Add the [JSON array][JsonArray] produced by the [builderAction] function to a resulting json object using the given [key].
+ *
+ * Returns the previous value associated with [key], or `null` if the key was not present.
+ */
+public fun JsonObjectBuilder.addJsonArray(key: String, builderAction: JsonArrayBuilder.() -> Unit): JsonElement? =
```
Comment (sandwwraith) [inline runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt:144]: It should be `putJsonArray`

File: runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt [hunk 2]
```diff
@@ -147,99 +117,143 @@ public class JsonObjectBuilder internal constructor() {
...
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.add(value: String?): Boolean = add(JsonPrimitive(value))
+
+/**
+ * Adds the [JSON][JsonObject] produced by the [builderAction] function to a resulting array.
+ *
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.putJsonObject(builderAction: JsonObjectBuilder.() -> Unit): Boolean =
```
Comment (sandwwraith) [inline runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt:247]: It should be `addJsonObject`

File: runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt [hunk 3]
```diff
@@ -11,41 +11,36 @@ import kotlin.contracts.*
...
- *     add("booleanKey", true)
- *     addArray("arrayKey") {
+ * val json = buildJsonObject {
+ *     put("booleanKey", true)
+ *     putJsonArray("arrayKey") {
  *         for (i in 1..10) add(i)
  *     }
- *     addJson("objectKey") {
+ *     putJsonObject("objectKey") {
  *         add("stringKey", "stringValue")
```
Comment (sandwwraith) [inline runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt:20]: change this documentation line to `put`

File: runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt [hunk 4]
```diff
@@ -147,99 +117,143 @@ public class JsonObjectBuilder internal constructor() {
...
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.add(value: String?): Boolean = add(JsonPrimitive(value))
+
+/**
+ * Adds the [JSON][JsonObject] produced by the [builderAction] function to a resulting array.
+ *
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.putJsonObject(builderAction: JsonObjectBuilder.() -> Unit): Boolean =
```
Comment (sandwwraith) [inline runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt:247]: should be `add`

File: runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt [hunk 5]
```diff
@@ -147,99 +117,143 @@ public class JsonObjectBuilder internal constructor() {
...
+    add(buildJsonObject(builderAction))
+
+/**
+ * Adds the [JSON][JsonArray] produced by the [builderAction] function to a resulting array.
+ *
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.putJsonArray(builderAction: JsonArrayBuilder.() -> Unit): Boolean =
```
Comment (sandwwraith) [inline runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt:255]: should be `add`

File: runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt (hunk 4 above)
Comment (qwwdfsad) [reply runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt:247]: sorry, my bad, should have checked the result of rename refactoring

**Pull request merged**